### DISCUSSION TITLE
1.1.0

### DISCUSSION THREAD

**Discussion by JohannesDeml** [JohannesDeml/NetworkBenchmarkDotNet#18] [General]
# 1.1.0 - .NET 6
* Update networking libraries
* Update Operating Systems
* Update .NET to 6.013 on Ubuntu and 6.0.15 on Windows

## Changes
* Upgrade NetCoreServer to v6.7.0
* Upgrade LietNetLib to 1.0.1
* Upgrade KCP to 1.34
* Upgrade BenchmarkDotNet to 0.13.5

## Results
![nbn-overview-1 1 0](https://user-images.githubusercontent.com/4518980/230493091-79e557c4-a74a-4064-a61b-05db06db56af.png)

### Hardware

* Ubuntu VPS
  * Virtual private server with dedicated CPU's running - [Hardware](https://www.netcup.eu/bestellen/produkt.php?produkt=2624)
  * Ubuntu 22.04.2 LTS x86-64 Kernel 5.15.0-48-generic
  
* Ubuntu Desktop / Windows Desktop
  * Desktop PC from 2020 - [Hardware](https://pcpartpicker.com/user/JohannesDeml/saved/zz7yK8)
  * Windows 11 Pro x86-64 Build 22621,1485 (22H2)
  * Ubuntu 22.04.2 LTS x86-64 Kernel 5.19.0-38-generic

### Software

* [.NET](https://dotnet.microsoft.com/download/dotnet) 6.0.13 on Ubuntu, 6.0.15 on Windows
* [BenchmarkDotNet](https://github.com/dotnet/BenchmarkDotNet) 0.13.5

<hr /><em>This discussion was created from the release <a href='https://github.com/JohannesDeml/NetworkBenchmarkDotNet/releases/tag/'>1.1.0</a>.</em>

**Comment by JohannesDeml** [comment]
@RevenantX I ran the benchmark now with .NET 6 and with the latest OS versions (and native sockets enabled), but sadly I didn't see any significant changes for LiteNetLib comapred to the last version. Can you check on your machine if the bahvior is the same? On windows it performs quite well and for reliable messages it performs better than ENet, but on Ubuntu with unreliable batched messages it is still very far away from the ENet results.
You can find all detailed results and logs in the release artifacts

**Reply by RevenantX** [reply]
Thanks! Seems that this is something linux specific. I will check benchmark on my linux machine

**Reply by JohannesDeml** [reply]
I just had another look what I did back then on a [different branch](https://github.com/JohannesDeml/NetworkBenchmarkDotNet/compare/master...cutting-edge), and noticed that I allowed unsafe code there:
```
    <PropertyGroup Condition=" '$(Configuration)' == 'Debug' ">
      <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
    </PropertyGroup>

    <PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
      <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
    </PropertyGroup>
```

Do you think this has a big impact on LiteNetLib? I guess it would make the comparison with ENet fairer, but I would like to only run the benchmark on all platforms again, if I'm certain that it is worth the time.

**Reply by RevenantX** [reply]
@JohannesDeml unsafe in latest releases (on nuget) is enabled by default. Also i checked performance of NativeSockets with unsafe too. But linux performance possibly worse because of another thing. I think this is because of different (between Windows and Linux) Blocking/NonBlocking implementation.
